#! /bin/sh /usr/share/dpatch/dpatch-run
## 05_checkwindows.dpatch 
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: check that windows exist in certain cases

@DPATCH@
diff -urNad xlockmore-5.27~/xlock/xlock.c xlockmore-5.27/xlock/xlock.c
--- xlockmore-5.27~/xlock/xlock.c	2008-08-31 22:09:56.000000000 -0400
+++ xlockmore-5.31/xlock/xlock.c	2009-02-22 14:14:28.023932384 -0500
@@ -2184,7 +2184,7 @@
 				if (event.xvisibility.state != VisibilityUnobscured) {
 					/* window was restacked or exposed */
 #ifndef __CYGWIN__
-					if (!debug && !inwindow)
+					if (!debug && !inwindow && event.xvisibility.window)
 						XRaiseWindow(dsp, event.xvisibility.window);
 #endif
 					call_refresh_hook((LockStruct *) NULL,
@@ -2202,7 +2202,8 @@
 				/* window config changed */
 #ifndef __CYGWIN__
 				if (!debug && !inwindow) {
-					XRaiseWindow(dsp, event.xconfigure.window);
+					if (event.xconfigure.window)
+						XRaiseWindow(dsp, event.xconfigure.window);
 					fixColormap(mode_info(dsp, screen, Scr[screen].window, False), ncolors,
 						saturation, mono, install, inroot, inwindow, verbose);
 				}
@@ -2838,7 +2839,7 @@
 
 			case VisibilityNotify:
 #ifndef __CYGWIN__
-				if (!debug && !inwindow) {
+				if (!debug && !inwindow && event.xany.window) {
 					XRaiseWindow(display, event.xany.window);
 				}
 #endif
@@ -2850,7 +2851,7 @@
 
 			case ConfigureNotify:
 #ifndef __CYGWIN__
-				if (!debug && !inwindow) {
+				if (!debug && !inwindow && event.xconfigure.window) {
 					XRaiseWindow(display, event.xconfigure.window);
 				}
 #endif
